### PR TITLE
ref(ui) Convert Saved Issue Search to new forms

### PR DESCRIPTION
**Pull request by markstory**
Update saved issue search to settings/forms. I've also replaced the custom footer buttons with more standard size ones, and replaced the alert with a component instead of LESS styles.

## Before

Screen Shot 2021-09-24 at 5 01 36 PM

## After

Screen Shot 2021-09-24 at 5 00 51 PM

### CODE REVIEW

File: static/app/views/issueList/createSavedSearchModal.tsx [hunk 1]
```diff
@@ -56,25 +48,25 @@ class CreateSavedSearchModal extends React.Component<Props, State> {
     return IssueSortOptions.DATE;
   }
 
-  onSubmit = (e: React.FormEvent) => {
+  onSubmit: OnSubmitCallback = (data, onSubmitSuccess, onSubmitError, event) => {
```
Comment (matejminar) [inline static/app/views/issueList/createSavedSearchModal.tsx:51]: nit:
```suggestion
  handleSubmit: OnSubmitCallback = (data, onSubmitSuccess, onSubmitError, event) => {
```

**Review by matejminar — APPROVED** [review]
👏

File: static/app/views/settings/components/forms/type.tsx [hunk 2]
```diff
@@ -198,3 +199,13 @@ export type JsonFormObject = {
...
   fields: FieldObject[];
 };
+
+export type Data = Record<string, any>;
+
+export type OnSubmitCallback = (
+  data: Data,
+  onSubmitSuccess: (data: Data) => void,
+  onSubmitError: (error: any) => void,
+  e: React.FormEvent,
```
Comment (priscilawebdev) [inline static/app/views/settings/components/forms/type.tsx:209]:
```suggestion
  event: React.FormEvent,
```
🤷‍♀️

File: static/app/views/issueList/createSavedSearchModal.tsx [hunk 3]
```diff
@@ -99,81 +92,69 @@ class CreateSavedSearchModal extends React.Component<Props, State> {
...
+            stacked
+            flexibleControlStateSize
             required
-            onChange={this.handleChangeQuery}
           />
           <SelectField
             key="sort"
             name="sort"
             label={t('Sort By')}
+            value={query}
```
Comment (priscilawebdev) [inline static/app/views/issueList/createSavedSearchModal.tsx:148]: do we need to pass the value here? doesn't the form do it for us?

File: static/app/views/issueList/createSavedSearchModal.tsx [hunk 4]
```diff
@@ -99,81 +92,69 @@ class CreateSavedSearchModal extends React.Component<Props, State> {
...
+    const initialData = {
+      name: '',
+      query,
+      sort: this.validateSortOption(sort),
+    };
 
     return (
-      <form onSubmit={this.onSubmit}>
+      <Form
+        onSubmit={this.onSubmit}
```
Comment (markstory) [inline static/app/views/issueList/createSavedSearchModal.tsx:111]:
```suggestion
        onSubmit={this.handleSubmit}
```

**Pull request merged**